lib/java/com/google/android/material/textfield/TextInputLayout.java

```diff
@@ -1014,6 +1014,27 @@ private void adjustFilledEditTextPaddingForLargeFont() {
     }
   }
 
+  /**
+   * Set the value to use for the EditText's collapsed top padding in box mode.
+   *
+   * @param boxCollapsedPaddingTop the value to use for the EditText's collapsed top padding
+   * @attr ref com.google.android.material.R.styleable#TextInputLayout_boxCollapsedPaddingTop
+   * @see #getBoxCollapsedPaddingTop()
+   */
+  public void setBoxCollapsedPaddingTop(int boxCollapsedPaddingTop) {
+    boxCollapsedPaddingTopPx = boxCollapsedPaddingTop;
```

Comment:
The value set here will be overridden by updateBoxCollapsedPaddingTop(). 

We will need a more sophisticated approach than this.

Reply:
Thank you for your comment. 

It seems that boxCollapsedPaddingTopPx inside updateBoxCollapsedPaddingTop is only updated when the font scaling is enabled. If we boxCollapsedPaddingTopPx is set via the style attribute, it will be also overriden by updateBoxCollapsedPaddingTop(). Seems that the behaviour is consistent if boxCollapsedPaddingTopPx is set in xml or via this new function

Reply:
Good point. Can you add some explanation that customized boxCollapsedPaddingTop will be disabled if the font scale is larger than 1.3?